Libplanet/Net/Transports/NetMQTransport.cs

```diff
@@ -813,96 +815,123 @@ private async Task ProcessRequest(MessageRequest req, CancellationToken cancella
                 DateTimeOffset.UtcNow - req.RequestedTime);
             DateTimeOffset startedTime = DateTimeOffset.UtcNow;
 
-            using var dealer = new DealerSocket(req.Peer.ToNetMQAddress());
-
-            _logger.Debug(
-                "Trying to send request {Message} {RequestId} to {Peer}...",
-                req.Message,
-                req.Id,
-                req.Peer);
-            var message = _messageCodec.Encode(
-                req.Message,
-                _privateKey,
-                AsPeer,
-                DateTimeOffset.UtcNow,
-                _appProtocolVersion);
-            var result = new List<Message>();
-            TaskCompletionSource<IEnumerable<Message>> tcs = req.TaskCompletionSource;
             try
             {
-                await dealer.SendMultipartMessageAsync(
-                    message,
-                    timeout: req.Timeout,
-                    cancellationToken: cancellationToken
-                );
+                long incrementedSocketCount = Interlocked.Increment(ref _socketCount);
+                _logger
+                    .ForContext("Tag", "Metric")
+                    .Debug(
+                    "{SocketCount} sockets open for processing request {Message} {RequestId}.",
+                    incrementedSocketCount,
+                    req.Message,
+                    req.Id);
+                using var dealer = new DealerSocket(req.Peer.ToNetMQAddress());
```

Comment:
Is it intended to `DealerSocket` move into `try` block?

Reply:
This is the **dumbest** way to ensure `_socketCount` stays valid. 🙄
Had problems `_socketCount` becoming negative for some reason.
I agree, structurally it's a mess. I've made another attempt to fix this.

Reply:
See the updated changes.